tests/test_change_oidc_provider.py

```diff
@@ -0,0 +1,37 @@
+import pytest
+
+from flask import url_for
+
+
+def request(url, method, data=None):
+    r = method(url, data=data)
+    return r
+
+
+@pytest.fixture
+def post(client):
+
+    def submit(url):
+
+        def response(data):
+            return request(url, client.post, data=data)
+
+        return response
+
+    return submit
+
+
+@pytest.fixture
+def post_switch_oidc_provider(post):
+    return post(url_for('base.switch_oidc_provider', caller='.admin'))
+
+
+class WhenUserChangesOIDCProvider(object):
+
+    def it_stores_new_oidc_provider_in_cookie(self,
+                                              post_switch_oidc_provider):
```

Comment:
Or, shorten `it_stores_new_oidc_provider_in_cookie` and `post_switch_oidc_provider` so it all fits on one line, maybe:
```python
def it_sets_idp_cookie(self, switch_idp):
```